src/mixxx.cpp

```diff
@@ -81,10 +81,41 @@
 #include "preferences/dialog/dlgprefmodplug.h"
 #endif
 
+#ifdef Q_OS_LINUX
+#include <QtGui/QX11Info>
+#include <X11/Xlib.h>
+#include <X11/Xlibint.h>
+// Xlibint.h predates C++ and defines macros which conflict
+// with references to std::max and std::min
+#undef max
+#undef min
+#endif
+
 namespace {
 
 const mixxx::Logger kLogger("MixxxMainWindow");
 
+// hack around https://gitlab.freedesktop.org/xorg/lib/libx11/issues/25
+// https://bugs.launchpad.net/mixxx/+bug/1805559
+#ifdef Q_OS_LINUX
+typedef Bool (*WireToErrorType)(Display*, XErrorEvent*, xError*);
+
+const int NUM_HANDLERS = 256;
+WireToErrorType __oldHandlers[NUM_HANDLERS] = {0};
```

Comment:
I still don't get the magic behind this array. Which unknown side effect sets the elements to a value other than 0? Currently the error interceptor does nothing except ignoring and dropping all incoming extension errors. This can be implemented in a much simpler way.

Reply:
Honestly, neither do I, nor do I understand the magic number 256. I've never worked with X11 code before, much less understand X's internals. I just copied and pasted from https://bugs.freedesktop.org/show_bug.cgi?id=59361#c4 and got it to build.

Reply:
256 because the error code is an unsigned char with values ranging from 0 to 255. But all slots are initialized with 0 (= `nullptr`) and are never updated.